Replicator/tests/ReplicatorCollectionSGTest.cc

```diff
@@ -722,6 +722,103 @@ TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Resolve Conflict SG", "[.SyncServe
     }
 }
 
+TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Auto Purge Enabled - Revoke Access - SGColl", "[.SyncServerCollection]") {
+    _authHeader = "Basic c2d1c2VyOnBhc3N3b3Jk"_sl; // "sguser:password" base64 encoded
+    slice docID = "apera-doc1"_sl;
+
+    // Setup pull filter:
+    _pullFilter = [](C4CollectionSpec collectionSpec, C4String docID, C4String revID,
+                     C4RevisionFlags flags, FLDict flbody, void *context) {
+        if ((flags & kRevPurged) == kRevPurged) {
+            ((ReplicatorAPITest*)context)->_counter++;
+            Dict body(flbody);
+            CHECK(body.count() == 0);
+        }
+        return true;
+    };
+
+    // One-shot pull setup
+    constexpr size_t collectionCount = 1;
+    std::array<C4CollectionSpec, collectionCount> collectionSpecs = {
+            Roses
+    };
+    std::array<C4Collection*, collectionCount> collections =
+            collectionPreamble(collectionSpecs, "sguser", "password");
+    std::array<C4ReplicationCollection, collectionCount> replCollections;
+    replCollections[0] = C4ReplicationCollection {
+        collectionSpecs[0], kC4Disabled, kC4OneShot,
+        nullslice, nullptr, _pullFilter, this
+    };
+    C4ParamsSetter paramsSetter = [&replCollections,&collectionCount](C4ReplicatorParameters& c4Params) {
+        c4Params.collectionCount = collectionCount;
+        c4Params.collections = replCollections.data();
+    };
+
+    // Setup onDocsEnded:
+    _enableDocProgressNotifications = true;
+    _onDocsEnded = [](C4Replicator* repl,
+                      bool pushing,
+                      size_t numDocs,
+                      const C4DocumentEnded* docs[],
+                      void* context) {
+        for (size_t i = 0; i < numDocs; ++i) {
+            auto doc = docs[i];
+            if ((doc->flags & kRevPurged) == kRevPurged) {
+                ((ReplicatorAPITest*)context)->_docsEnded++;
+            }
+        }
+    };
+
+    auto collRoses = collections[0];
+
+    // Put doc in remote DB, in channels a and b
+    sendRemoteRequest("PUT", docID.asString(), R"({"channels":["a", "b"]})");
+
+    // Pull doc into CBL:
+    C4Log("-------- Pulling");
+    replicate(paramsSetter);
```

Comment:
Not sure how you managed server db. I guess you empty the db before this test. Otherwise, you will pull all the docs from the db. I used docIDs to manage it.

Reply:
Ah yeah, that's a good point... I'll change it to use docIDs and also think I will make the test randomly generate docID at the beginning of the test so that previous test runs can never interfere.

Reply:
"idPrefix" in the existing tests just does it.